aten/src/ATen/detail/MPSHooksInterface.h

```diff
@@ -28,7 +28,7 @@ struct TORCH_API MPSHooksInterface {
     return false;
   }
 
-  virtual const Generator& getDefaultMPSGenerator(C10_UNUSED DeviceIndex device_index = -1) const {
+  virtual const Generator& getDefaultMPSGenerator() const {
```

Comment:
Hmm, I think it should take the device index, but error out if index is other than `0` (as `index` is inherent property of the device and it's existence is not checked when device is created)

Reply:
I removed it because there was no use case for it. I can add it but it'll be used only in a check like `TORCH_CHECK(device_index == 0, ...)`. 
The other options is to add that later when we support multiple devices.